api/maketdata.go

```diff
@@ -123,7 +127,21 @@ func getTickersHandler(storage storage.Market) func(c *gin.Context) {
 			if err != nil {
 				continue
 			}
-			r.ApplyRate(md.Currency, rate.Rate, rate.PercentChange24h)
+			if r.Price.Currency != blockatlas.DefaultCurrency {
+				newRate, err := storage.GetRate(strings.ToUpper(r.Price.Currency))
+				if err == nil {
+					exchangeRate *= newRate.Rate
+					percentChange = newRate.PercentChange24h
+				} else {
+					tickerRate, err := storage.GetTicker(strings.ToUpper(r.Price.Currency), "")
+					if err == nil {
+						exchangeRate *= tickerRate.Price.Value
+						percentChange = big.NewFloat(tickerRate.Price.Change24h)
+					}
+				}
+			}
```

Comment:
this conversion logic is quite a bit of complexity for the controller. How do you feel about moving this in some conversion utility?